### DISCUSSION TITLE
Add Image Versioning

### DISCUSSION THREAD

**Discussion by juancarlosjr97** [directus/directus#3995] [Feature Requests]
The image versioning is not working.

1. The issue and what you expected to happen
    
An image modified, i.e., rotated, the reverted change using the versioning should revert to the original rotation 

2. Exact steps to reproduce this issue:

- Upload an image
- Rotate the image
- Click on `revisions` and revert to the version named `Created`
- The image should be still be rotated

3. Your environment:
- AWS RDS PostgreSQL
- Directus v9.0.0-rc.34

4. Any other relevant information we might need to reproduce this issue
    - The name of the file is reverted if it is changed but no the image rotation, crop, etc

**Comment by benhaynes** [comment]
Not really a bug, more of a feature request, as Directus never had this... but I'll leave that for @rijkvanzanten to decide. I agree it would be a great feature to add (in Discussions), shouldn't be too hard either... just need to agree on the ideal storage naming/location.

**Comment by juancarlosjr97** [comment]
Thank you for the reply @benhaynes. I have not tested this on v8, so it is good to know that it would a feature. I would have expected on the versioning to restore also the image previous state.

Probably, could be done using S3 versioning if it is enabled.

**Comment by benhaynes** [comment]
Yeah, it seems intuitive to have files be versioned and it's definitely something we want to add. We can't use anything specific to S3 since all features need to be abstracted away so they are not storage adapter specific (eg: needs to work on local filesystem too). I assume that this will be a new naming convention in the **upload** directory... perhaps with these centralized in a sub-directory to keep things clean.

**Comment by rijkvanzanten** [comment]
Yeah we'd have to store a copy of the file _before_ it was updated, and rename it in a way that lets us identify the revision it goes with 🤔

**Reply by zicklag** [reply]
I haven't looked at the source, so maybe this doesn't make sense, but maybe files in the database could have a `latest_revision` integer column and then the files on disk would just be named `${uuid}-${revision}.${extension}`.

Then we shoudn't need to rename anything, if I'm not missing something obvious. :)

**Comment by zicklag** [comment]
Just wanted to add that this would be useful to my team, if Directus continues to satisfy our other needs and we keep using it.

I've just gotten started with Directus as a backend for my website, and as I try to simplify my team's cloud infrastructure, and I realized that it could probably replace our Nextcloud instance as well, because we mostly used Nextcloud as a file share. File revisions is probably the biggest thing Directus doesn't have that was useful in Nextcloud.

As an additional, thought, if we have file revisions, they are going to take way more space than normal edits to text fields, so we probably want a way to delete old revisions, and maybe even use Flows to create customized, automatic revision cleanup policies. That would be awesome.

I'm loving Directus so far, and I'm quite impressed with it's capability and interface. Keep up the good work!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
